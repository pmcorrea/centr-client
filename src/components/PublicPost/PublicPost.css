.PublicPost {
  background-color: white;
  color: black;
  width: 90vw;
  position: relative;
  margin: 10px auto;
}

.PublicPost__title {
  font-size: 30px;
  margin: 0 0 10px 0;
  max-width: 80%;
  color: black;
}
.PublicPost__title a {
  color: #cacaca;
  display: block;
  text-decoration: none;
}

.PublicPost__dates {
  bottom: 8px;
  color: #a7a5a5;
  font-size: 14px;
  position: relative;
  margin-top: 10px;
}

.PublicPostRoute {
  padding: 20px 40px;
}

p {
  color: black;
}

.back_button {
  font-size: 60px;
  font-weight: bold;
  margin: 0px;
  margin-bottom: 10px;
  padding: 0px;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  color: rebeccapurple;
}

.back_button:hover {
  color: orange;
}

.public_post_avatar {
  width: 80px;
  height: 80px;
  border-radius: 50%;
  box-shadow: 0px 5px 5px rgba(128, 128, 128, 0.733);
}
