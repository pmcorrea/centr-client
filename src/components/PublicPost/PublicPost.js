import React, { Component } from "react";
// Context and service
import MainContext from "../../contexts/MainContext";
import AuthApiService from "../../services/auth-api-service.js"
// Style
import "./PublicPost.css";

export default class PublicPost extends Component {
	static contextType = MainContext;

	handleDeletePost(value) {
		AuthApiService.handleDeletePost(value)
		.then(result => {
			this.context.getDataWithToken()
		})
		.catch(error => {
			console.error(error)
		})  
	}


	render() {
		let someDateString = this.props.modified 
		let someDate = new Date(someDateString)
		someDate = someDate.toLocaleString()

		return (
			<div className="PublicPost">
				<div className="PublicPost_container">
					<img src={this.props.avatar} alt="avatar" className="public_post_avatar"></img>

					<h2 className="PublicPost__title">
						{this.props.name}
					</h2>

					<div className="PublicPost__dates">
						<div className="PublicPost__dates-modified">
							<span className="PublicPost_Date">
								{someDate}
							</span>
						</div>
					</div>
				</div>
			</div>
		);
	}
}
