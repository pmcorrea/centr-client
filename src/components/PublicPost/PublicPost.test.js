import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PublicPost from './PublicPost'

describe(`PublicPost component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    modified: new Date(2018, 12, 15),
  }

  it('renders a .PublicPost by default', () => {
    const wrapper = shallow(<PublicPost />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the PublicPost given props', () => {
    const wrapper = shallow(<PublicPost {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
