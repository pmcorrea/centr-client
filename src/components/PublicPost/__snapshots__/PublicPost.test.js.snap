// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicPost component renders a .PublicPost by default 1`] = `
<div
  className="PublicPost"
>
  <div
    className="PublicPost_container"
  >
    <img
      alt="avatar"
      className="public_post_avatar"
    />
    <h2
      className="PublicPost__title"
    />
    <div
      className="PublicPost__dates"
    >
      <div
        className="PublicPost__dates-modified"
      >
        <span
          className="PublicPost_Date"
        >
          Invalid Date
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`PublicPost component renders the PublicPost given props 1`] = `
<div
  className="PublicPost"
>
  <div
    className="PublicPost_container"
  >
    <img
      alt="avatar"
      className="public_post_avatar"
    />
    <h2
      className="PublicPost__title"
    >
      test-class-name
    </h2>
    <div
      className="PublicPost__dates"
    >
      <div
        className="PublicPost__dates-modified"
      >
        <span
          className="PublicPost_Date"
        >
          1/15/2019, 12:00:00 AM
        </span>
      </div>
    </div>
  </div>
</div>
`;
